### DISCUSSION TITLE
Adding/Removing components while iterating

### DISCUSSION THREAD

**Discussion by BuBuMan** [skypjack/entt#1171] [General]
Hey @skypjack! Your posts at https://skypjack.github.io/ have been very educational. I am learning and having a blast reading through them. I am not sure if this is the right place to ask a question about the posts, but I noticed that entt supports adding/removing components dynamically at runtime which is a very nice feature. On one of your posts, it is mentioned one way to achieve this is to iterate backwards instead of forward, but I have been wondering how you were able to achieve this in the following case:

- Assuming pools are using dynamically growing arrays, how do you avoid the cases where a reallocation happens due to an addition?

Thank you for your time

**Comment by BuBuMan** [comment]
I think I got my answer by reading more. Seems like pools are stored in pages instead :)

**Comment by skypjack** [comment]
Yup, that's because of the pagination. If you enable pointer stability, you can also remove elements without invalidating pointers. Otherwise, the swap-and-pop approach can cause troubles in this case.
